components/LayoutTransformControl/src/LayoutTransformControl.cs

```diff
@@ -0,0 +1,466 @@
+    {
+        // Save existing content and remove it from the visual tree
+        FrameworkElement savedContent = Child!;
+        Child = null;
+
+        // Apply new template
+        base.OnApplyTemplate();
+
+        // Find template parts
+        if (GetTemplateChild("LayoutRoot") is Panel panel)
+        {
+            _layoutRoot = panel;
+        }
+
+        if (GetTemplateChild("MatrixTransform") is MatrixTransform matrixTransform)
+        {
+            _matrixTransform = matrixTransform;
+        }
+
+        // RestoreAsync saved content
+        Child = savedContent;
+
+        // Apply the current transform
+        TransformUpdated();
+    }
+
+    /// <summary>
+    /// Notifies the LayoutTransformControl that some aspect of its Transform property has changed.
+    /// </summary>
+    /// <remarks>
+    /// Call this to update the LayoutTransform in cases where
+    /// LayoutTransformControl wouldn't otherwise know to do so.
+    /// </remarks>
+    public void TransformUpdated()
+    {
+        ProcessTransform();
+    }
+
+    /// <summary>
+    /// Return true if Size a is smaller than Size b in either dimension.
+    /// </summary>
+    /// <param name="a">The left size.</param>
+    /// <param name="b">The right size.</param>
+    /// <returns>A value indicating whether the left size is smaller than
+    /// the right.</returns>
+    private static bool IsSizeSmaller(Size a, Size b)
+    {
+        // WPF equivalent of following code:
+        // return ((a.Width < b.Width) || (a.Height < b.Height));
+        return (a.Width + AcceptableDelta < b.Width) || (a.Height + AcceptableDelta < b.Height);
+    }
+
+    /// <summary>
+    /// Processes the current transform to determine the corresponding
+    /// matrix.
+    /// </summary>
+    private void ProcessTransform()
+    {
+        // Get the transform matrix and apply it
+        _transformation = GetTransformMatrix(Transform).Round(DecimalsAfterRound);
+
+        if (_matrixTransform != null)
+        {
+            _matrixTransform.Matrix = _transformation;
+        }
+
+        // New transform means re-layout is necessary
+        InvalidateMeasure();
+    }
+
+    /// <summary>
+    /// Walks the Transform and returns the corresponding matrix.
+    /// </summary>
+    /// <param name="transform">The transform to create a matrix for.
+    /// </param>
+    /// <returns>The matrix calculated from the transform.</returns>
+    private Matrix GetTransformMatrix(Transform transform)
+    {
+        if (transform != null)
+        {
+            // WPF equivalent of this entire method (why oh why only WPF...):
+            // return transform.Value;
+
+            // Process the TransformGroup
+            var transformGroup = transform as TransformGroup;
+
+            if (transformGroup != null)
+            {
+#if HAS_UNO
+                var groupMatrix = Windows.UI.Xaml.Media.Matrix.Identity;
+#else
+                var groupMatrix = Matrix.Identity;
+#endif
```

Comment:
Think there's still something off here?

```
         D:\a\Windows\Windows\components\LayoutTransformControl\src\LayoutTransformControl.cs(160,35): error CS0234: The type or namespace name 'Media' does not exist in the namespace 'Windows.UI.Xaml' (are you missing an assembly reference?) [D:\a\Windows\Windows\components\LayoutTransformControl\src\CommunityToolkit.WinUI.Controls.LayoutTransformControl.csproj]
         D:\a\Windows\Windows\components\LayoutTransformControl\src\LayoutTransformControl.cs(214,16): error CS0234: The type or namespace name 'Media' does not exist in the namespace 'Windows.UI.Xaml' (are you missing an assembly reference?) [D:\a\Windows\Windows\components\LayoutTransformControl\src\CommunityToolkit.WinUI.Controls.LayoutTransformControl.csproj]
```

I think the Uno line has to be split for UWP vs. WINAPPSDK? Or maybe @jeromelaban's comment means he thinks we can just remove the conditional all together. Did you try just having the original code of `Matrix.Identity`?

Otherwise, could be easier to put this whole block to redefine what `Matrix` is at the top, so the code here and below just remains `Matrix.Identity`? Then it's easy to see the using and remove it later if fixed elsewhere in Uno later?

e.g.
```cs
#if HAS_UNO && WINUI3
     using Matrix = Microsoft.UI.Xaml.Media.Matrix;
 #else
    using Matrix = Windows.UI.Xaml.Media.Matrix
#endif
```